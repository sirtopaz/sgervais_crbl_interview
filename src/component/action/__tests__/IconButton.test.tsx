import { vi } from "vitest";
import { render, screen } from "@testing-library/react";

import { ActionIcon, IconButton } from "..";
import userEvent from "@testing-library/user-event";

const clickHandlerMock = vi.fn();

describe("COMPONENT:ACTION:IconButton", () => {
  it("should render a proper HTML with supplied props", () => {
    render(
      <IconButton
        icon={ActionIcon.OPEN}
        a11yLabel="TESTING_LABEL"
        onClick={clickHandlerMock}
      />
    );

    const button = screen.getByLabelText("TESTING_LABEL");

    expect(button).toBeInstanceOf(HTMLButtonElement);
    expect(button).toHaveClass("icon-button");
  });

  it("should have deterministic HTML", () => {
    // as a controlled visual component the same input props should always generate the same HTML structure

    const { container } = render(
      <IconButton
        icon={ActionIcon.CLOSED}
        a11yLabel="TESTING_LABEL"
        onClick={clickHandlerMock}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should handle click", async () => {
    render(
      <IconButton
        icon={ActionIcon.OPEN}
        a11yLabel="TESTING_LABEL"
        onClick={clickHandlerMock}
      />
    );

    const button = screen.getByLabelText("TESTING_LABEL");
    await userEvent.click(button);

    expect(clickHandlerMock).toHaveBeenCalled();
  });
});
