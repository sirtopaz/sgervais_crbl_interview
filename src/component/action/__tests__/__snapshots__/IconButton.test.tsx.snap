// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`COMPONENT:ACTION:IconButton > should have deterministic HTML 1`] = `
<div>
  <button
    aria-label="TESTING_LABEL"
    class="icon-button"
    type="button"
  >
    <svg
      aria-hidden="true"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <title>
        TESTING_LABEL
      </title>
      <path
        d="M9 5L15 12L9 19"
      />
    </svg>
  </button>
</div>
`;
